### ISSUE TITLE
Please Provide Support for Rasperry PI 5 Device

### ISSUE THREAD

**Issue by Suhailakl** [nisargjhaveri/WirelessAndroidAutoDongle#123]

**Comment by nisargjhaveri** [comment]
Looks like the upstream buildroot still does not have RPi 5 support. We can try to add support, I believe there have been some proposed patches as well.

Unfortunately, I don't have a RPi 5 yet to experiment on. Need to wait for someone to add support upstream first.

**Comment by nisargjhaveri** [comment]
Seems like the the buildroot recently added support for RPi5, but that is still not part of any stable release. I believe it should be part of the 2024.08 release when that is available. Also, I don't currently own a RPi 5, so someone else needs to test the setup once we actually bump the buildroot submodule.

**Comment by nisargjhaveri** [comment]
Another thing to consider would be the power requirements of a RPi 5 and would it be able to work well on usb power from the car.

**Comment by mylife4aiurr** [comment]
I can help test pi5. Should I install the pi4 image on pi5?

(pi5 is over kill for this application, why not use a pi 0. Also I doubt my cars head unit can deliver 5v @ 5A)

**Comment by nisargjhaveri** [comment]
Pi 4 image most likely won't work on Pi 5. To build for Pi 5, we'd essentially need appropriate defconfig file, something similar to [raspberrypi4_defconfig](https://github.com/nisargjhaveri/WirelessAndroidAutoDongle/blob/main/aa_wireless_dongle/configs/raspberrypi4_defconfig).

Upstream buildroot contains one right now. See [raspberrypi5_defconfig](https://github.com/buildroot/buildroot/blob/master/configs/raspberrypi5_defconfig). Having a defconfig will help build for the board. Next would be to test and see if we need any more customisations for it to work.

**Comment by se2crid** [comment]
any updates?

**Comment by mylife4aiurr** [comment]
> any updates?

Nope. It's doesn't make sense for several reasons to port this to pi5. 
Cost:
Pi5 cost significantly more a pi0 w or pi2 w. - why waste money. 

Power requirements:
Pi5 requires a USB-C power supply that can offer the full 5V 5A at 27 watts - that power requirement exceeds most car USB capability. 

Why waste developers time on something more expensive and less practical?

**Comment by se2crid** [comment]
> > any updates?
> 
> Nope. It's doesn't make sense for several reasons to port this to pi5. Cost: Pi5 cost significantly more a pi0 w or pi2 w. - why waste money.
> 
> Power requirements: Pi5 requires a USB-C power supply that can offer the full 5V 5A at 27 watts - that power requirement exceeds most car USB capability.
> 
> Why waste developers time on something more expensive and less practical?

If you already have a pi 5 it is not really a waste and the 25 watts is only if you use every USB port and the Pcie port

**Comment by sree3333** [comment]
I have a Pi 5, you let me know on how to test it?

**Comment by sree3333** [comment]
@manio Could you describe me in detail how to build for raspberry pi 5 with buildroot
I have tried and I got a sdimage but it is not working after I flashed it to the card . @nisargjhaveri

**Comment by manio** [comment]
The config has to be adjusted, but what you mean that it is not working? Can you connect HDMI and see on the console what is the error exactly?

**Comment by sree3333** [comment]
@manio @nisargjhaveri Stuck here, did I do the flash wrong?

![Image](https://github.com/user-attachments/assets/7c047673-aa79-4ca5-9558-c9d0caab3b96)

**Comment by manio** [comment]
You need to clone `WirelessAndroidAutoDongle` with:
`$ git clone --recurse-submodules https://github.com/nisargjhaveri/WirelessAndroidAutoDongle`

Then you need to use this [file](https://skyboo.net/temp/aa/raspberrypi5_defconfig). As the new platform is not yet configured/prepared, so the easiest would be to overwrite eg `raspberrypi4_defconfig` and build it all like for pi4.

Build instruction is [here](https://github.com/nisargjhaveri/WirelessAndroidAutoDongle/blob/main/BUILDING.md).

**Comment by sree3333** [comment]
@manio package/pkg-generic.mk:257: *** BR2_GLOBAL_PATCH_DIR contains nonexistent directory board/raspberrypi/patches.  Stop.

**Comment by manio** [comment]
You sure you compiling it properly? The directory is [there](https://github.com/nisargjhaveri/WirelessAndroidAutoDongle/tree/main/aa_wireless_dongle/patches), and rest of the configs have the same value.

I suggest you first try to compile it without any changes to see if you have a proper dongle image as the result of the build process...

**Comment by sree3333** [comment]
Ok, I will try to build with the rpi4 original config first and then change to try to rpi5 @manio

**Comment by sree3333** [comment]
@manio the build is successful for rpi5 , but no visible bluetooth to connect

**Comment by sree3333** [comment]
@manio @nisargjhaveri I think there is some issue with the rpi5 defconfig 

"+ mv /home/mark-42/Projects/WirelessAndroidAutoDongle/buildroot/output/rpi5/target/etc/aawgd.conf /home/mark-42/Projects/WirelessAndroidAutoDongle/buildroot/output/rpi5/images/aawgd.conf
mv: cannot stat '/home/mark-42/Projects/WirelessAndroidAutoDongle/buildroot/output/rpi5/target/etc/aawgd.conf': No such file or directory"

Its not able to make aawgd.conf file while make -j10, I tried copying the aawgd.conf from raspberry pi 4 build but that does'nt seem to work

**Comment by manio** [comment]
And for RPi4 it is creating three?

**Comment by sree3333** [comment]
yes

**Comment by manio** [comment]
Maybe you have some build error you didn't notice?

**Comment by sree3333** [comment]
Could you check once from your side as well? @manio

**Comment by sree3333** [comment]
The code ran successfully, I even ran it for multiple variantions and even using docker and the terminal. All gives a same sdimage file of 159.4 MB. @manio 

![Image](https://github.com/user-attachments/assets/9e210c30-4192-4e10-bb5b-5323d9f9f889)
This is how it will look like after the successful completion without any error right?

**Comment by sree3333** [comment]
You want the complete log? @manio

**Comment by manio** [comment]
Ok trying to build it (https://github.com/manio/WirelessAndroidAutoDongle/tree/pi5)...
